src/__tests__/credentials-autoAccept.test.ts

```diff
@@ -0,0 +1,874 @@
+      })
+
+      expect(JsonTransformer.toJSON(aliceCredentialRecord)).toMatchObject({
+        createdAt: expect.any(Date),
+        offerMessage: {
+          '@id': expect.any(String),
+          '@type': 'https://didcomm.org/issue-credential/1.0/offer-credential',
+          comment: 'some comment about credential',
+          credential_preview: {
+            '@type': 'https://didcomm.org/issue-credential/1.0/credential-preview',
+            attributes: [
+              {
+                name: 'name',
+                'mime-type': 'text/plain',
+                value: 'John',
+              },
+              {
+                name: 'age',
+                'mime-type': 'text/plain',
+                value: '99',
+              },
+            ],
+          },
+          'offers~attach': expect.any(Array),
+        },
+        state: CredentialState.OfferReceived,
+      })
+
+      // below values are not in json object
+      expect(aliceCredentialRecord.id).not.toBeNull()
+      expect(aliceCredentialRecord.getTags()).toEqual({
+        threadId: aliceCredentialRecord.threadId,
+        state: aliceCredentialRecord.state,
+        connectionId: aliceConnection.id,
+      })
+      expect(aliceCredentialRecord.type).toBe(CredentialRecord.name)
+
+      testLogger.test('Faber waits for credential request from Alice')
+      faberCredentialRecord = await waitForCredentialRecord(faberAgent, {
+        threadId: aliceCredentialRecord.threadId,
+        state: CredentialState.RequestReceived,
+      })
+
+      testLogger.test('Alice waits for credential from Faber')
+      aliceCredentialRecord = await waitForCredentialRecord(aliceAgent, {
+        threadId: faberCredentialRecord.threadId,
+        state: CredentialState.CredentialReceived,
+      })
+
+      testLogger.test('Faber waits for credential ack from Alice')
+      faberCredentialRecord = await waitForCredentialRecord(faberAgent, {
+        threadId: faberCredentialRecord.threadId,
+        state: CredentialState.Done,
+      })
+
+      expect(aliceCredentialRecord).toMatchObject({
+        type: CredentialRecord.name,
+        id: expect.any(String),
+        createdAt: expect.any(Date),
+        offerMessage: expect.any(Object),
+        requestMessage: expect.any(Object),
+        metadata: { requestMetadata: expect.any(Object) },
+        credentialId: expect.any(String),
+        state: CredentialState.Done,
+      })
+
+      expect(faberCredentialRecord).toMatchObject({
+        type: CredentialRecord.name,
+        id: expect.any(String),
+        createdAt: expect.any(Date),
+        offerMessage: expect.any(Object),
+        requestMessage: expect.any(Object),
+        state: CredentialState.Done,
+      })
+    })
+  })
+
+  describe('Auto accept on `contentApproved`', () => {
+    beforeAll(async () => {
+      ;({ faberAgent, aliceAgent, credDefId, schemaId, faberConnection, aliceConnection } = await setupCredentialTests(
```

Comment:
weird formatting

Reply:
When trying to return multiple variables from a function and putting them into a `let` this is the only way how. I can't say I am a fan.

Reply:
But we don't have to return variables here? I'm talking about the `;({`.

```suggestion
      { faberAgent, aliceAgent, credDefId, schemaId, faberConnection, aliceConnection } = await setupCredentialTests(
```

Reply:
It's because the method call is wrapped inside `()`

Reply:
the brackets are necessary for using the return values in the object.

`setupCredentialTests` returns multiple variables inside an object which I have to bind into a higher level `let`. This is only possible with wrapping it inside the `()`.

Without these brackets the following error is thrown: `Parsing error: Declaration or statement expected. This '=' follows a block of statements, so if you intended to write a destructuring assignment, you might need to wrap the the whole assignment in parentheses.`.

solution was found here: [SO](https://stackoverflow.com/questions/35576307/declaration-or-statement-expected-javascript-typescript)

Reply:
Ah I didn't knew. Sorry

Weird...